File: common/src/main/java/dev/obscuria/fragmentum/common/signal/_Signal1.java
package dev.obscuria.fragmentum.common.signal;

import dev.obscuria.fragmentum.v2.api.common.signal.Signal1;

public final class _Signal1<P1> extends _Signal<Signal1.Listener<P1>> implements Signal1<P1> {

    @Override public void emit(P1 p1) {
        emitInternal(it -> it.consume(p1));
    }
}

File: api/src/main/kotlin/dev/obscuria/fragmentum/tools/signal/Signal1.kt
@file:Suppress("unused", "UNUSED_PARAMETER", "RedundantNullableReturnType")

package dev.obscuria.fragmentum.tools.signal

class Signal1<P1> : Signal<Signal1.Listener<P1>>() {

    fun emit(p1: P1): Unit = error("API stub")

    fun interface Listener<P1> {

        fun consume(p1: P1)
    }
}